commcare_connect/opportunity/views.py

```diff
@@ -1376,3 +1363,308 @@ def sync_deliver_units(request, org_slug, opp_id):
         message = "Failed to retrieve updates. No available build at the moment."
 
     return HttpResponse(content=message, status=status)
+
+
+@org_member_required
+def user_visit_verification(request, org_slug, opp_id, pk):
+    opportunity = get_opportunity_or_404(opp_id, org_slug)
+    opportunity_access = get_object_or_404(OpportunityAccess, opportunity=opportunity, pk=pk)
+    is_program_manager = request.org.program_manager and (
+        (request.org_membership != None and request.org_membership.is_admin) or request.user.is_superuser  # noqa: E711
+    )
+
+    user_visit_counts = UserVisit.objects.filter(opportunity_access=opportunity_access).aggregate(
+        pending=Count("id", filter=Q(status=VisitValidationStatus.pending)),
+        rejected=Count("id", filter=Q(status=VisitValidationStatus.rejected)),
+        flagged=Count(
+            "id", filter=Q(flagged=True, status__in=[VisitValidationStatus.approved, VisitValidationStatus.rejected])
+        ),
+        total=Count("*"),
+    )
+
+    visits = UserVisit.objects.filter(opportunity_access=opportunity_access)
+    flagged_info = defaultdict(lambda: {"name": "", "approved": 0, "rejected": 0})
+    for visit in visits:
+        for flag in visit.flags:
+            if visit.status in (VisitValidationStatus.approved, VisitValidationStatus.rejected):
+                flagged_info[flag][visit.status] += 1
+                flagged_info[flag]["name"] = flag
+    flagged_info = flagged_info.values()
+    last_payment_details = Payment.objects.filter(opportunity_access=opportunity_access).order_by("-date_paid").first()
+
+    response = render(
+        request,
+        "opportunity/user_visit_verification.html",
+        context={
+            "header_title": "Worker",
+            "opportunity_access": opportunity_access,
+            "counts": user_visit_counts,
+            "flagged_info": flagged_info,
+            "last_payment_details": last_payment_details,
+            "MAPBOX_TOKEN": settings.MAPBOX_TOKEN,
+            "opportunity": opportunity_access.opportunity,
+            "is_program_manager": is_program_manager,
+        },
+    )
+    return response
+
+
+def get_user_visit_counts(opportunity_access_id: int, date):
+    opportunity_access = OpportunityAccess.objects.get(id=opportunity_access_id)
+    visit_count_kwargs = dict(approved=Count("id", filter=Q(status=VisitValidationStatus.approved)))
```

Comment:
move to else?

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/571/commits/373167b842e2298f6b2fd66e15731346af759bc8